shop/src/main/java/ro/msg/learning/shop/service/ProductService.java

```diff
@@ -0,0 +1,103 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.exception.EntityNotFoundException;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.ProductCategory;
+import ro.msg.learning.shop.model.Supplier;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+    private final ProductCategoryRepository categoryRepository;
+    private final SupplierRepository supplierRepository;
+    private final ProductRepository productRepository;
+
+    public ProductService(
+        ProductCategoryRepository categoryRepository,
+        SupplierRepository supplierRepository,
+        ProductRepository productRepository
+    ) {
+        this.categoryRepository = categoryRepository;
+        this.supplierRepository = supplierRepository;
+        this.productRepository = productRepository;
+    }
+
+    public ProductCategory saveProductCategory(ProductCategory productCategory) {
+        return categoryRepository.save(productCategory);
+    }
+
+    public Optional<ProductCategory> findProductCategoryById(Long categoryId) {
+        return categoryRepository.findById(categoryId);
+    }
+
+    public List<ProductCategory> findAllProductCategories() {
+        return categoryRepository.findAll();
+    }
+
+    @Transactional
+    public void deleteProductCategoryById(Long categoryId) {
+        if (categoryRepository.existsById(categoryId)) {
+            categoryRepository.deleteById(categoryId);
+        }
+    }
+
+    public Supplier saveSupplier(Supplier supplier) {
+        return supplierRepository.save(supplier);
+    }
+
+    public Optional<Supplier> findSupplierById(Long supplierId) {
+        return supplierRepository.findById(supplierId);
+    }
+
+    public List<Supplier> findAllSuppliers() {
+        return supplierRepository.findAll();
+    }
+
+    @Transactional
+    public void deleteSupplierById(Long supplierId) {
+        if (supplierRepository.existsById(supplierId)) {
+            supplierRepository.deleteById(supplierId);
+        }
+    }
+
+    @Transactional
+    public Product saveProduct(Product product) {

```

Comment:
1. null check
2. you could make use of the separation of concern principle and refactor this into more methods, such as a method that handles category and one that handles supplier. 
3. The services communicate with one another, and each service with its corresponding repo. The supplier and category logic need to be in their own services (goes hand in hand with point 2)